Source/Core/Editor/XRUtils/OpenXRPackageEnabler.cs

```diff
@@ -26,27 +28,127 @@ internal sealed class OpenXRPackageEnabler : XRProvider
 
         protected override string XRLoaderName { get; } = "OpenXRLoader";
 
-        protected override void InitializeXRLoader(object sender, EventArgs e)
+        private const int MaxRetries = 3;
+
+        protected override async void InitializeXRLoader(object sender, EventArgs e)
         {
             BuilderProjectSettings settings = BuilderProjectSettings.Load();
-
             BuildTargetGroup buildTargetGroup = BuildTargetGroup.Standalone;
-            FeatureHelpers.RefreshFeatures(buildTargetGroup);
 
-            foreach (string controllerProfileType in settings.OpenXRControllerProfiles)
+            // Wait for OpenXR to be ready
+            if (!await WaitForOpenXRSettings(buildTargetGroup))
             {
-                OpenXRFeature feature = OpenXRSettings.Instance.GetFeatures<OpenXRInteractionFeature>().FirstOrDefault(f => f.GetType().Name == controllerProfileType);
-
-                if (feature != null)
+                UnityEngine.Debug.LogError("[VR Builder] OpenXRSettings could not be accesed. Controller profiles are not enabled. Try manually enabling profiles in Project Settings > XR Plug-in Management > OpenXR.");
+            }
+            else
+            {
+                // Enable controller profiles
+                foreach (string controllerProfileType in settings.OpenXRControllerProfiles)
                 {
-                    feature.enabled = true;
+                    await EnableFeatureWithRetry<OpenXRInteractionFeature>(controllerProfileType, buildTargetGroup);
                 }
+
+                EnableWorkaroundForOXRB656(buildTargetGroup);
             }
 
             AssetDatabase.SaveAssets();
-
             base.InitializeXRLoader(sender, e);
         }
+
+        /// <summary>
+        /// Enable WorkaroundForOXRB656
```

Comment:
Since this is an exceptional case, I would be more clear on why we are trying to enable this thing.